packages/core/src/storages/request_queue_v2.ts

```diff
@@ -0,0 +1,988 @@
+        }));
+
+        const request = requestLike instanceof Request
+            ? requestLike
+            : new Request(requestLike);
+
+        const cacheKey = getRequestId(request.uniqueKey);
+        const cachedInfo = this.requestCache.get(cacheKey);
+
+        if (cachedInfo) {
+            request.id = cachedInfo.id;
+
+            return {
+                wasAlreadyPresent: true, // We may assume that if request is in local cache then also the information if the
+                // request was already handled is there because just one client should be using one queue.
+                wasAlreadyHandled: cachedInfo.isHandled,
+                requestId: cachedInfo.id,
+                uniqueKey: cachedInfo.uniqueKey,
+            };
+        }
+
+        const queueOperationInfo = await this.client.addRequest(request, { forefront }) as RequestQueueV2OperationInfo;
+        queueOperationInfo.uniqueKey = request.uniqueKey;
+
+        const { requestId, wasAlreadyPresent } = queueOperationInfo;
+        this._cacheRequest(cacheKey, queueOperationInfo);
+
+        if (!wasAlreadyPresent && !this.requestIdsInProgress.has(requestId) && !this.recentlyHandledRequestsCache.get(requestId)) {
+            this.assumedTotalCount++;
+
+            // Performance optimization: add request straight to head if possible
+            this._maybeAddRequestToQueueHead(requestId, forefront);
+        }
+
+        return queueOperationInfo;
+    }
+
+    /**
+     * Fetches URLs from requestsFromUrl and returns them in format of list of requests
+     */
+    protected async _fetchRequestsFromUrl(source: InternalSource): Promise<RequestOptions[]> {
+        const { requestsFromUrl, regex, ...sharedOpts } = source;
+
+        // Download remote resource and parse URLs.
+        let urlsArr;
+        try {
+            urlsArr = await this._downloadListOfUrls({ url: requestsFromUrl, urlRegExp: regex, proxyUrl: await this.proxyConfiguration?.newUrl() });
+        } catch (err) {
+            throw new Error(`Cannot fetch a request list from ${requestsFromUrl}: ${err}`);
+        }
+
+        // Skip if resource contained no URLs.
+        if (!urlsArr.length) {
+            this.log.warning('list fetched, but it is empty.', { requestsFromUrl, regex });
+            return [];
+        }
+
+        return urlsArr.map((url) => ({ url, ...sharedOpts }));
+    }
+
+    /**
+     * @internal wraps public utility for mocking purposes
+     */
+    private async _downloadListOfUrls(options: DownloadListOfUrlsOptions): Promise<string[]> {
+        return downloadListOfUrls(options);
+    }
+
+    /**
+     * Adds all fetched requests from a URL from a remote resource.
+     */
+    protected async _addFetchedRequests(source: InternalSource, fetchedRequests: RequestOptions[], options: RequestQueueV2OperationOptions) {
+        const { requestsFromUrl, regex } = source;
+        const { processedRequests } = await this.addRequests(fetchedRequests, options);
```

Comment:
this needs to use the new `addRequestsBatched` method, please check the changes I did in the RQ in last month, some of them are clearly not here

https://github.com/apify/crawlee/commits/master/packages/core/src/storages/request_queue.ts